general/administration/hardware-acceleration.md

```diff
@@ -87,6 +87,127 @@ services:
       # RPi 4
       - /dev/vchiq:/dev/qchiq
 ```
+## Debian Docker Nvidia Hardware acceleration:
+
+In order to achieve Hardware decode inside docker containers, some steps are needed:
+Thoses steps where wrote for Debian Stretch.
+
+Prerequisits:
+https://github.com/nvidia/nvidia-docker/wiki/Installation-(version-2.0)
+
+    GNU/Linux x86_64 with kernel version > 3.10
+    Docker >= 1.12
+    NVIDIA GPU with Architecture > Fermi (2.1)
+    NVIDIA drivers ~= 361.93 (untested on older versions)
+
+Double check your GPU show on debian:
+```lspci | grep VGA```
+
+Update your host:
+```apt-get update && apt-get dist-upgrade -y```
+
+Install curl:
+```apt-get install curl```
+
+Edit sources.list in /ect/apt/sources.list:
+Add "non-free contrib" to every sources already configured when needed:
+
+```deb http://ftp.ch.debian.org/debian/ stretch main```
+
+become
+
+```deb http://ftp.ch.debian.org/debian/ stretch main non-free contrib```
+
+Add sources for Nvidia-container:
+```
+curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add
+distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
+curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list | sudo tee /etc/apt/sources.list.d/nvidia-docker.list
+```
+Update your package list:
+```apt-get update```
+
+Install linux-headers:
+```apt-get install linux-headers-$(uname -r|sed 's/[^-]*-[^-]*-//')```
+Else to use stretch-backports:
+```apt-get install -t stretch-backports linux-headers-$(uname -r|sed 's/[^-]*-[^-]*-//')```
+
+Install Nvidia docker2:
+```apt-get install nvidia-docker2```
+When prompted to choose to keep or install the maintainer package file say "y" to install maintainer version
+
+OPTIONAL: After install you may want to add nvidia as default runtime: editing /etc/docker/daemon.json like this:
+```
+{
+    "default-runtime": "nvidia",
+    "runtimes": {
+        "nvidia": {
+            "path": "nvidia-container-runtime",
+            "runtimeArgs": []
+        }
+    }
+}
+```
+Restart docker services
+```sudo pkill -SIGHUP docker```
+
+Install nvidia drivers and dependencies:
+```apt-get install -t stretch-backports nvidia-driver libnvcuvid1 libnvidia-encode1 libcuda1 nvidia-smi```
+
+Reboot your host:
+```reboot now```
+
+Validate your driver and docker are correctly setted up
+Driver test from host and docker side:
+```
+nvidia-smi
+docker run --gpus 0 nvidia/cuda:9.0-base nvidia-smi
+```
+Validate access to needed ressources from host and docker:
+```
+ldconfig -p | grep cuvid
+ldconfig -p | grep libnvidia-encode.so.1
+```
+Start your Jellyfin container adding thoses environement parameters:
+```
+-e "NVIDIA_DRIVER_CAPABILITIES=all" \
+-e NVIDIA_VISIBLE_DEVICES=all \
+--runtime=nvidia \
+--gpus all \
+```
+A complete run command would looks like:
+```
+docker run -d \
+--name=jellyfin \
+-e PUID=1000 \
+-e PGID=100 \
```

Comment:
Can you run id in your container? AFAIK, the jellyfin image doesn't support user environmental variables.

Reply:
My bad - this can be removed.
Was usefull when using linuxserver.io container.

Reply:
so if you want to run the user in the container use --user 1000:1000

you might have to add the user to the video group? depends on your os and who owns the gfx cards. for my rpi4, i had to do 

`usermod -aG video user`

Reply:
This was not needed on my side to add the user to any group
As the env. settings for the user was not correctly set i guess it was running as root so i had no problem.
Can add infos about this.

Reply:
sure. any additional info you can add will always be welcome. I dont have an nvidia card so I rely on others for the documentation :)